corgi/core/models.py

```diff
@@ -679,6 +679,18 @@ def get_latest_components(
                 return Component.objects.none()
             return Component.objects.filter(Q(ROOT_COMPONENTS_CONDITION & query)).using(using)
 
+    @property
+    def provides_queryset(self, using: str = "read_only") -> QuerySet["Component"]:
+        """Returns unique aggregate "provides" for the latest components in this stream,
+        for use in templates"""
+        unique_provides = (
+            self.get_latest_components()
+            .distinct("provides__pk")
+            .values_list("provides__pk")
```

Comment:
I think it's best to use flat=True and put .distinct() after the .values_list(), like below:
```suggestion
            .values_list("provides__pk", flat=True)
            .distinct()
```

Reply:
Good idea. I also added order_by and iterator here.